src/Http/Http.Abstractions/src/IHttpContextFactory.cs

```diff
@@ -5,9 +5,22 @@
 
 namespace Microsoft.AspNetCore.Http
 {
+    /// <summary>
+    /// Provides methods to create <see cref="HttpContext"/>
+    /// </summary>
     public interface IHttpContextFactory
     {
+        /// <summary>
+        /// Creates a <see cref="HttpContext"/> instance for each request.
+        /// </summary>
+        /// <param name="featureCollection">The collection of HTTP features to set on the created instance.</param>
+        /// <returns>The <see cref="HttpContext"/> instance.</returns>
         HttpContext Create(IFeatureCollection featureCollection);
+
+        /// <summary>
+        /// Releases the unmanaged resources and disposes of the managed resources used by the <see cref="HttpContext"/>.
```

Comment:
```suggestion
        /// Releases resources held by the <see cref="HttpContext"/>.
```

Dispose is a lot less rigidly defined now, let's just say it "releases resources" instead of being specific about unmanaged/managed/etc.